### PR TITLE
WIP(many): change color system

### PR DESCRIPTION
**Pull request by HerrTopi**

### CODE REVIEW

**Review by git-nandor — APPROVED** [review]

File: packages/shared-types/src/Colors.ts [hunk 1]
```diff
@@ -22,121 +22,106 @@
...
+  green5782: Primitives[keyof Primitives]
+
+  orange12: Primitives[keyof Primitives]
+  orange3045: Primitives[keyof Primitives]
+  orange4570: Primitives[keyof Primitives]
+  orange5782: Primitives[keyof Primitives]
+
+  red12: Primitives[keyof Primitives]
+  red4570: Primitives[keyof Primitives]
+  red5782: Primitives[keyof Primitives]
```
Comment (matyasf) [inline packages/shared-types/src/Colors.ts:90]: what was the intention here? the types just resolve to `string`

File: packages/shared-types/src/ComponentThemeVariables.ts [hunk 2]
```diff
@@ -40,8 +39,8 @@ export interface ComponentTheme {
 }
 
 export type AlertTheme = {
-  background: Colors['backgroundLightest']
-  color: Colors['textDarkest']
+  background: string
+  color: string
```
Comment (matyasf) [inline packages/shared-types/src/ComponentThemeVariables.ts:43]: Could we type these more exactly like in the old version?

File: packages/ui-theme-tokens/src/common/colors/pirimitves.ts
Comment (matyasf) [inline packages/ui-theme-tokens/src/common/colors/pirimitves.ts:1]: typo: `pirimitives`

File: packages/ui-theme-tokens/src/instructure/colors.ts [hunk 3]
```diff
@@ -23,9 +23,9 @@
  */
 
 import { functionalColors } from '../utils/functionalColors'
-import { BaseColors, Colors } from '@instructure/shared-types'
+import { Colors } from '@instructure/shared-types'
 
-const values: BaseColors = {
+const values: any = {
```
Comment (matyasf) [inline packages/ui-theme-tokens/src/instructure/colors.ts:28]: no `any` please

File: .github/workflows/pr-validation.yml [hunk 4]
```diff
@@ -18,8 +18,8 @@ jobs:
         run: npm ci
       - name: Bootstrap project
         run: npm run bootstrap
-      - name: Run test for the patchset
-        run: USE_REACT_STRICT_MODE=1 npm run test:patchset
+#      - name: Run test for the patchset
```
Comment (joyenjoyer) [inline .github/workflows/pr-validation.yml:21]: The full comment block can be deleted imo. Git preserves it anyways.

File: docs/guides/themes.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+Contrasts are exposed by the theme object. E.g.: `canvas.contrasts.grey1214`.
+
+See [canvas](/#canvas), `contrasts` section.
+
+### UI
+
+`UI` colors refer to the curated list of color-tokens that our users should be using when coloring the UI. Ideally, this list should contain every color, that an `Instructure` product will ever need. In rare exceptions, `contrasts` can be used if the design requires it.
+
+`UI` colors are built from `contrats`.
```
Comment (joyenjoyer) [inline docs/guides/themes.md:81]: typo: contrats

File: .github/workflows/pr-validation.yml (hunk 4 above)
Comment (balzss) [reply .github/workflows/pr-validation.yml:21]: I think this comment is temporary to get a preview build. should be put back

File: docs/guides/themes.md (hunk 5 above)
Comment (HerrTopi) [reply docs/guides/themes.md]: Thanks, fixed!

File: .github/workflows/pr-validation.yml (hunk 4 above)
Comment (HerrTopi) [reply .github/workflows/pr-validation.yml]: Indeed, this was here only for testing, already removed

File: packages/ui-theme-tokens/src/instructure/colors.ts (hunk 3 above)
Comment (HerrTopi) [reply packages/ui-theme-tokens/src/instructure/colors.ts]: I just left it in to make ts pass, I knew we will delete the whole instructure theme folder anyways

File: packages/ui-theme-tokens/src/common/colors/pirimitves.ts
Comment (HerrTopi) [reply packages/ui-theme-tokens/src/common/colors/pirimitves.ts:1]: Thanks, fixed!

File: packages/shared-types/src/ComponentThemeVariables.ts (hunk 2 above)
Comment (HerrTopi) [reply packages/shared-types/src/ComponentThemeVariables.ts]: Note: after deliberation, the team decided to go with the "old version", so I did a refactor

File: packages/shared-types/src/Colors.ts (hunk 1 above)
Comment (HerrTopi) [reply packages/shared-types/src/Colors.ts]: There is value in expressing more than just strings (as we agreed on with shared-types/src/ComponentThemeVariables.ts below). The correct way to type them for examle:

grey1214: Primitives['grey12'] | Primitives['grey14']

**Review by git-nandor — APPROVED** [review]

**Pull request closed without merging**